### PR TITLE
poppler: update to 22.02.0

### PR DESCRIPTION
**Pull request by michaelld**
#### Description

update poppler to 22.02.0. I've been running updated versions for a long time, with no issues.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.6.4 20G507
Xcode 12.3 12C33

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by evanmiller** [comment]
@michaelld I headed down a similar path but abandoned the effort. Here are the changes I made, cherry-pick what you like!

```
-PortGroup           legacysupport 1.0
+PortGroup           legacysupport 1.1
+
+# strndup
+legacysupport.newest_darwin_requires_legacy 10
```

```
-compiler.cxx_standard 2014
+compiler.cxx_standard 2017
```

```
@@ -80,6 +80,7 @@ configure.args-append \
                     -DBUILD_QT5_TESTS=OFF \
                     -DBUILD_QT6_TESTS=OFF \
                     -DBUILD_CPP_TESTS=OFF \
+                    -DRUN_GPERF_IF_PRESENT=OFF \
                     -DWITH_NSS3=ON \
                     -DUSE_BOOST_HEADERS=OFF
```

**Review by mascguy — APPROVED** [review]

**Comment by michaelld** [comment]
@evanmiller  Thanks for those. Let me check those out & see what makes sense to integrate

**Comment by michaelld** [comment]
I went ahead and reviewed the various new requirements & tweaked the Portfile to meet those. Let's see how CI does here. I'll aim to merge this Friday morning (US/Eastern) if nobody else has and there are no other objections, per maintainer timeout.

**Comment by michaelld** [comment]
All CI passes. Merging per maintainer timeout.

**Comment by evanmiller** [comment]
I believe this can be closed then: https://trac.macports.org/ticket/63539

**Comment by evanmiller** [comment]
Also it looks like High Sierra and earlier builds are failing:

```
/bin/sh: GLIB2_MKENUMS_PYTHON-NOTFOUND: command not found
make[2]: *** [glib/poppler-enums.h] Error 127
```

https://build.macports.org/builders/ports-10.13_x86_64-builder/builds/142016

**Comment by michaelld** [comment]
drat

**Comment by michaelld** [comment]
great catch! I need to add Python to the dependencies & patch this `find` to work correctly. More recent macOS provides `/usr/bin/python3` which is being used here; not available in somewhat older macOS / OSX.

**Comment by michaelld** [comment]
9354f8f38a6a032cebf0e53a3d4823532c8b24a5

File: graphics/poppler/Portfile
Comment (evanmiller) [inline graphics/poppler/Portfile:60]: Is this supposed to be `2011`? AFAIK there is no C 2001.

Comment (michaelld) [reply graphics/poppler/Portfile:60]: Correct! My bad ... 7e30f6e6d8376b24530a6d052fbfde11ddb8473a

**Pull request merged**
